base/utils/R/ensemble.R

```diff
@@ -191,96 +214,194 @@ write.ensemble.configs <- function(defaults, ensemble.samples, settings, model,
+    new.state<-restart$new.state
+    ensemble.id<-restart$ensemble.id
     
-    # create folders (cleaning up old ones if needed)
-    if (clean) {
-      unlink(file.path(settings$rundir, run.id))
-      unlink(file.path(settings$modeloutdir, run.id))
+    for (i in seq_len(settings$ensemble$size)) {
+      do.call(my.write_restart, 
+              args =  list(outdir = settings$host$outdir, 
+                           runid = run.id[[i]], 
+                           start.time = restart$start.time,
+                           stop.time =restart$stop.time, 
+                           settings = settings,
+                           new.state = new.state[i, ], 
+                           new.params = new.params[[i]], 
+                           inputs =list(met=list(path=inputs$samples[[i]])), 
+                           RENAME = TRUE)
+      )
     }
-    dir.create(file.path(settings$rundir, run.id), recursive = TRUE)
-    dir.create(file.path(settings$modeloutdir, run.id), recursive = TRUE)
-    
-    # write run information to disk
-    cat("runtype     : ensemble\n",
-        "workflow id : ", workflow.id, "\n",
-        "ensemble id : ", ensemble.id, "\n",
-        "run         : ", counter, "/", settings$ensemble$size, "\n",
-        "run id      : ", run.id, "\n",
-        "pft names   : ", as.character(lapply(settings$pfts, function(x) x[['name']])), "\n",
-        "model       : ", model, "\n",
-        "model id    : ", settings$model$id, "\n",
-        "site        : ", settings$run$site$name, "\n",
-        "site  id    : ", settings$run$site$id, "\n",
-        "met data    : ", settings$run$site$met, "\n",
-        "start date  : ", settings$run$start.date, "\n",
-        "end date    : ", settings$run$end.date, "\n",
-        "hostname    : ", settings$host$name, "\n",
-        "rundir      : ", file.path(settings$host$rundir, run.id), "\n",
-        "outdir      : ", file.path(settings$host$outdir, run.id), "\n",
-        file = file.path(settings$rundir, run.id, "README.txt"))
-    
-    do.call(my.write.config, args = list(
-      defaults = defaults, 
-      trait.values = lapply(
-        ensemble.samples, function(x, n) { x[n, , drop=FALSE] }, n=counter
-      ), 
-      settings = settings, 
-      run.id = run.id)
-    )
-    cat(run.id, file = file.path(settings$rundir, "runs.txt"), sep = "\n", append = TRUE)
+    params<-new.params
+    return(invisible(list(runs = data.frame(id=run.id), ensemble.id = ensemble.id, samples=list(met=inputs)
+                          )
+                     ))
   }
-
-  return(invisible(list(runs = runs, ensemble.id = ensemble.id)))
+  
+  
+  
 } # write.ensemble.configs
+
+
+
+#' Function for reurning the met ensuble based on sampling method, parent or etc
+#'
+#' @param settings list of PEcAn settings
+#' @param method Method for sampling - For now looping or sampling with replacement is implemented
+#' @param parenids If met's ids are read from a parent what are the ids
```

Comment:
Is there are reason to call this `parenids` rather than `parent_ids`? The latter is much easier to understand. Also, because every db table has IDs, clarify exactly what IDs you need here (input? dbfile? ensemble?). Finally, no need to reference 'met' specifically if this has been generalized.